plugins/ltac/rewrite.ml

```diff
@@ -1927,10 +1924,7 @@ let build_morphism_signature env sigma m =
   in
   let morph = e_app_poly env evd (PropGlobal.proper_type env) [| t; sig_; m |] in
   let evd = solve_constraints env !evd in
-  let evd = Evd.minimize_universes evd in
-  let m = Evarutil.nf_evars_universes evd (EConstr.Unsafe.to_constr morph) in
-  Pretyping.check_evars env evd (EConstr.of_constr m);
-  Evd.evar_universe_context evd, m
+  evd, morph
```

Comment:
Is it obvious why the `check_evars` can now be skipped?

Reply:
Actually there is a minor difference, `DeclareDef` calls `check_evars_are_solved` whereas this code calls check_evars; I'm not sure what's going on here; see also #11984 .

IMO we should use a single call for all paths, unless a special check is needed, then I'd suggest we turn it into a parameter. (cc: @SkySkimmer )

Reply:
Other than that, I think semantics are the same.